src/core/plugin_loader.py

```diff
@@ -36,16 +37,24 @@ def load(directory="plugins", prefix="plugins", permissive=False):
             plugins.append(plugin)
             module_name = "{0}.{1}.plugin_settings".format(prefix, dir)
 
+            # Load settings module
+            plugin_settings = import_module(module_name)
+
             # Load hooks
-            hooks.append(load_hooks(module_name))
+            hooks.append(load_hooks(plugin_settings))
 
             # Check for workflow
-            workflow_check = check_plugin_workflow(module_name)
+            workflow_check = check_plugin_workflow(plugin_settings)
             if workflow_check:
                 settings.WORKFLOW_PLUGINS[workflow_check] = module_name
+                STAGE_CHOICES.append(
+                    (plugin_settings.STAGE, plugin_settings.PLUGIN_NAME)
+                )
+                ELEMENT_STAGES[
```

Comment:
I don't think mutating a module constant is a good idea... I like using register decorators for this kind of task, which let you control when the registers are applied at runtime instead of depending on import order